pkg/sql/internal.go

```diff
@@ -1088,6 +1088,12 @@ func (ie *InternalExecutor) execInternal(
 			}); err != nil {
 			return nil, err
 		}
+		if err := stmtBuf.Push(ctx, Sync{
```

Comment:
no relevant to your change but wondering why `if len(qargs) == 0 ` can be used determine if it's the last and only statement in the batch. Isn't it just checking if there're any arguments or placeholders?

Reply:
good question! the internal executor sits in a weird place where it needs to craft these pgwire messages itself. len(qargs) is used to determine if it should use the simple protocol or the extended protocol. (the extended protocol is essentially synonymous with using a prepared statement with placeholders.)

we know it's the last in the batch because the internal executor always sends just one query per batch.